Add a dummy user with uid 1000

cbliard opened the pull request:
I find the usage of `fig` fantastic and created aliases to run ember easily:

```
alias ember='fig --file ~/.ember-cli.fig.yml run --rm ember'
```

Yet, files created with `ember new my-app` are all owned by root. To prevent this, I created a new image from yours with a dummy user having the same uid as me. I modified the fig.yml file to run as this dummy user:

```
# Copied from https://registry.hub.docker.com/u/geoffreyd/ember-cli/
# Allows using ember-cli without installing npm and bower

ember: &defaults
  image: cbliard/ember-cli-custom
  user: dummy
  volumes:
    - .:/usr/src/app #if your on docker actual

server:
  <<: *defaults
  command: server --watcher polling
  ports:
    - 4200:4200
    - 35729:35729

npm:
  <<: *defaults
  entrypoint: ['/usr/local/bin/npm']

bower:
  <<: *defaults
  entrypoint: ['/usr/local/bin/bower', '--allow-root']
```

Now when I run `ember new my-app`, the created files have the expected owner/group.

User mapping is planned for Docker, but for the moment it has been a useful hack for me.

cbliard committed:
Add a dummy user with uid 1000

Docker 1.2 cannot map the root user of a container to a non-root user of
the Docker host. It is a planned functionality. So we create a dummy user
with a uid matching the host user. Docker can then be run with this dummy
user and created files will be owned by the host user.

geoffreyd commented:
Ahh, cool ... I've not run this directly on linux, only through boot2docker with NFS shares, so I had not seen this behaviour.

Thanks for catching this @cbliard 

Not all distros use the same UID for 'first user', some are 101, some are 501 and others 1000 ... Any thoughts on how we can make this configurable?

cbliard commented:
I'll try to find a way to make it configurable. Should be possible, but hacky.

cbliard committed:
Allow any uid/gid with USER_UID & USER_GID env var

If USER_UID and USER_GID are defined, ember will be run as a user with
these uid/gid, so the created files will not be owned by root.

Usage with `docker`:

    docker run -e USER_UID=1000 -e USER_GID=1000 <image>

Usage with `fig`, in fig.yml file:

    ember:
      image: <image>
      environment:
        - USER_UID=1000
        - USER_GID=1000

cbliard commented:
Now it runs a little `as_user` script which changes the user uid/gid if USER_UID and USER_GID environment variables are defined. Thus the fig.yml looks like this

```
# Copied from https://registry.hub.docker.com/u/geoffreyd/ember-cli/
# Allows using ember-cli without installing npm and bower

ember: &defaults
  image: cbliard/test
  environment:
    - USER_UID=1000
    - USER_GID=1000
  volumes:
    - .:/usr/src/app #if you're on docker actual

server:
  <<: *defaults
  command: server --watcher polling
  ports:
    - 4200:4200
    - 35729:35729

npm:
  <<: *defaults
  entrypoint: ['/usr/local/bin/as_user', '/usr/local/bin/npm']

bower:
  <<: *defaults
  entrypoint: ['/usr/local/bin/as_user', '/usr/local/bin/bower', '--allow-root']
```

If run without defining USER_UID/USER_GID, the behavior is the same as before. So "Upgrading" the image should not break anything.

cbliard commented:
Too old and obsolete. Closing.
